src/main/kotlin/dev/fff/fractalizer/fitnessfunction/http/HttpCheckFitnessFunctionHandler.kt

```diff
@@ -20,22 +20,22 @@ class HttpCheckFitnessFunctionHandler(val httpClient: OkHttpClient) : FitnessFun
             LOGGER.warn("Expected fitness function ${fitnessFunction.name} to have properties")
             return false
         }
-        fitnessFunction.properties["url"] ?: run {
+        val url = fitnessFunction.properties["url"] ?: run {
             LOGGER.warn("Expected fitness function ${fitnessFunction.name} to have a property 'url'")
             return false
         }
 
         try {
             val response = httpClient.newCall(Request.Builder()
-                    .url(fitnessFunction.properties["url"])
+                    .url(url)
                     .build()).execute()
 
             if (!response.isSuccessful) {
                 return false
             }
-
-            return fitnessFunction.properties["expectedResult"]?.contains(response.body().string()) ?: true
-        } catch (e: ConnectException) {
+            //FIXME !! added for compile error
+            return fitnessFunction.properties["expectedResult"]?.contains(response.body()!!.string()) ?: true
```

Comment:
When expectedResult is not null and response body is null the function should evaluate to false. The !! will generate a null pointer exception.